src/api/wallet/WalletApi.ts

```diff
@@ -457,6 +462,26 @@ export class WalletApiImpl implements WalletApi {
     await this._notifyListeners()
   }
 
+  public async getGasPrice(): Promise<number | null> {
+    // this never errors
+    // returns undefined if unable to fetch
+    let gasPrice = await this._fetchGasPrice()
+
+    if (gasPrice) return +gasPrice
+    try {
+      // fallback to gasPrice from provider
+      // {"jsonrpc":"2.0","method":"eth_gasPrice"} call
+      gasPrice = await this._web3.eth.getGasPrice()
```

Comment:
We have `api/gasStation`, and it depends on WalletApi
1. I don't want to init it separately and pass walletApi to it. Since result of `fetchGasPriceFactory(walletApi)` needs to go inside `walletApi.connect` to be used in composed provider
2. Previously that kind of interdependence caused cyclic deps.
3. [L353](https://github.com/gnosis/dex-react/blob/313518ef08fb26eb809195fa07010a22b4e6d49b/src/api/wallet/WalletApi.ts#L353) keeps dependencies flatter

Reply:
Aha! that logic is only if the gas API fails. Then all good.

I'm not a fan of using this kind of fallbacks, at least by default, cause it can bring bugs that are very difficult to reproduce. Gas price estimator is an important service for the app, and I would be OK assuming it work (or fail, or we handle the case)

Anyways, in this case, it's not a super strong opinion